src/MvvmNano.Forms/MvvmNanoFormsPresenter.cs

```diff
@@ -119,7 +132,16 @@ public void SetApplication(Application application)
             IView view = CreateViewFor<TViewModel>();
             view.SetViewModel(viewModel);
 
-            return OpenPageAsync(view as Page);
+            return StartOpeningPageAsync<TViewModel>(view as Page);
+        }
+
+        /// <summary>
+        /// Opens the page a <see cref="MasterDetailData.ViewModelType"/> is referencing to.
+        /// </summary>
+        /// <param name="data"></param>
+        public void SetDetail(MasterDetailData data)
+        {
+            typeof(MvvmNanoFormsPresenter).GetRuntimeMethod("NavigateToViewModel", parameters: new Type[0]).MakeGenericMethod(data.ViewModelType).Invoke(this, (object[])null);
```

Comment:
`NavigateTo<TViewModel>()` is calling `SetDetail` if the page is a detail at the moment.
The logic for that is here (MvvmNanoFormsPresenter line 230):

        private async Task StartOpeningPageAsync<TViewModel>(Page page)
        { 
            var masterDetailApplication = (MvvmNanoMasterDetailApplication)Application;

            if (IsMasterDetailApplication()
                && masterDetailApplication.MasterPage != null 
                && masterDetailApplication.MasterDetails.FirstOrDefault(o => o.ViewModelType == typeof (TViewModel)) != null) //Check if the new page is a detail page
            {
                //Check if the current page is opened as modal page and close it if that is the case.
                if (masterDetailApplication.MasterPage.Detail.Navigation.ModalStack.Any()
                    && masterDetailApplication.MasterPage.Detail.Navigation.ModalStack.FirstOrDefault() == CurrentPage)
                {
                    await masterDetailApplication.MasterPage.Detail.Navigation.PopModalAsync();
                }
                masterDetailApplication.MasterPage.SetDetail(page);
            } 
            else
                await OpenPageAsync(page); 
        }

I could: 

1. Add `SetDetail<TViewModel>()` and leave the logic above intact.
2. Add `SetDetail<TViewModel>()` and remove the logic above.
3. Leave everything as it is.